### PR TITLE
git-lfs: fix build on darwin

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

- **git-lfs: format**
- **git-lfs: move to by-name**
- **git-lfs: minor improvements**
- **git-lfs: disable network tests on darwin**

`git-lfs` fails to build on `darwin` with the sandbox enabled.
Unfortunately, enabling `__darwinAllowLocalNetworking` does not help.
I disabled all the failing tests.

I also moved the package to `by-name`. Don't pay attention to the removal of `lowPrio` in `all-packages.nix`.
It has originally been introduced because `git-lfs1` was a thing.

cc @twey

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361797`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 31 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>chirpstack-concentratord</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dub-to-nix</li>
    <li>envision</li>
    <li>git-lfs</li>
    <li>github-backup</li>
    <li>github-backup.dist</li>
    <li>kcl</li>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>makehuman</li>
    <li>nim_lk</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>nix-update.dist</li>
    <li>npins</li>
    <li>nvfetcher</li>
    <li>outline</li>
    <li>prefetch-yarn-deps</li>
    <li>sparkleshare</li>
    <li>typescript-language-server</li>
    <li>update-nix-fetchgit</li>
    <li>update-python-libraries</li>
    <li>vimPluginsUpdater</li>
    <li>yarn2nix</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 31 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>chirpstack-concentratord</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dub-to-nix</li>
    <li>envision</li>
    <li>git-lfs</li>
    <li>github-backup</li>
    <li>github-backup.dist</li>
    <li>kcl</li>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>makehuman</li>
    <li>nim_lk</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>nix-update.dist</li>
    <li>npins</li>
    <li>nvfetcher</li>
    <li>outline</li>
    <li>prefetch-yarn-deps</li>
    <li>sparkleshare</li>
    <li>typescript-language-server</li>
    <li>update-nix-fetchgit</li>
    <li>update-python-libraries</li>
    <li>vimPluginsUpdater</li>
    <li>yarn2nix</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>update-python-libraries</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 26 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dub-to-nix</li>
    <li>git-lfs</li>
    <li>github-backup</li>
    <li>github-backup.dist</li>
    <li>kcl</li>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>makehuman</li>
    <li>nim_lk</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>nix-update.dist</li>
    <li>npins</li>
    <li>nvfetcher</li>
    <li>prefetch-yarn-deps</li>
    <li>typescript-language-server</li>
    <li>update-nix-fetchgit</li>
    <li>vimPluginsUpdater</li>
    <li>yarn2nix</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>update-python-libraries</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 26 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dub-to-nix</li>
    <li>git-lfs</li>
    <li>github-backup</li>
    <li>github-backup.dist</li>
    <li>kcl</li>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>makehuman</li>
    <li>nim_lk</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>nix-update.dist</li>
    <li>npins</li>
    <li>nvfetcher</li>
    <li>prefetch-yarn-deps</li>
    <li>typescript-language-server</li>
    <li>update-nix-fetchgit</li>
    <li>vimPluginsUpdater</li>
    <li>yarn2nix</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
The failure of `update-python-libraries` is surely unrelated:
```
patching script interpreter paths in /nix/store/kj697m2n0la1bsd143apf7hg0blb61pw-update-python-libraries
/nix/store/kj697m2n0la1bsd143apf7hg0blb61pw-update-python-libraries: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/krnn565sxwdryj68a1y102y8k94igpgg-python3-3.12.7-env/bin/python3"
sed: couldn't open temporary file /nix/store/sedflgxs3: Operation not permitted
```

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361797`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 31 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>chirpstack-concentratord</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dub-to-nix</li>
    <li>envision</li>
    <li>git-lfs</li>
    <li>github-backup</li>
    <li>github-backup.dist</li>
    <li>kcl</li>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>makehuman</li>
    <li>nim_lk</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>nix-update.dist</li>
    <li>npins</li>
    <li>nvfetcher</li>
    <li>outline</li>
    <li>prefetch-yarn-deps</li>
    <li>sparkleshare</li>
    <li>typescript-language-server</li>
    <li>update-nix-fetchgit</li>
    <li>update-python-libraries</li>
    <li>vimPluginsUpdater</li>
    <li>yarn2nix</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 31 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>chirpstack-concentratord</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dub-to-nix</li>
    <li>envision</li>
    <li>git-lfs</li>
    <li>github-backup</li>
    <li>github-backup.dist</li>
    <li>kcl</li>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>makehuman</li>
    <li>nim_lk</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>nix-update.dist</li>
    <li>npins</li>
    <li>nvfetcher</li>
    <li>outline</li>
    <li>prefetch-yarn-deps</li>
    <li>sparkleshare</li>
    <li>typescript-language-server</li>
    <li>update-nix-fetchgit</li>
    <li>update-python-libraries</li>
    <li>vimPluginsUpdater</li>
    <li>yarn2nix</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>update-python-libraries</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 26 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dub-to-nix</li>
    <li>git-lfs</li>
    <li>github-backup</li>
    <li>github-backup.dist</li>
    <li>kcl</li>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>makehuman</li>
    <li>nim_lk</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>nix-update.dist</li>
    <li>npins</li>
    <li>nvfetcher</li>
    <li>prefetch-yarn-deps</li>
    <li>typescript-language-server</li>
    <li>update-nix-fetchgit</li>
    <li>vimPluginsUpdater</li>
    <li>yarn2nix</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>update-python-libraries</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 26 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dub-to-nix</li>
    <li>git-lfs</li>
    <li>github-backup</li>
    <li>github-backup.dist</li>
    <li>kcl</li>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>makehuman</li>
    <li>nim_lk</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>nix-update.dist</li>
    <li>npins</li>
    <li>nvfetcher</li>
    <li>prefetch-yarn-deps</li>
    <li>typescript-language-server</li>
    <li>update-nix-fetchgit</li>
    <li>vimPluginsUpdater</li>
    <li>yarn2nix</li>
  </ul>
</details>

**Comment by niklaskorz** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361797`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 27 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dub-to-nix</li>
    <li>git-lfs</li>
    <li>github-backup</li>
    <li>github-backup.dist</li>
    <li>kcl</li>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>makehuman</li>
    <li>nim_lk</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nix-update-source.dist</li>
    <li>nix-update.dist</li>
    <li>npins</li>
    <li>nvfetcher</li>
    <li>prefetch-yarn-deps</li>
    <li>typescript-language-server</li>
    <li>update-nix-fetchgit</li>
    <li>update-python-libraries</li>
    <li>vimPluginsUpdater</li>
    <li>yarn2nix</li>
  </ul>
</details>

File: pkgs/by-name/gi/git-lfs/package.nix [hunk 1]
```diff
@@ -71,6 +102,8 @@ buildGoModule rec {
     updateScript = nix-update-script { };
   };
 
+  __darwinAllowLocalNetworking = true;
```
Comment (niklaskorz) [inline pkgs/by-name/gi/git-lfs/package.nix:105]: Are there any remaining tests that need this, with the networking tests being disabled above?

Comment (GaetanLepage) [reply pkgs/by-name/gi/git-lfs/package.nix:105]: Yes, it fails with another error if I remove it.

**Review by niklaskorz — APPROVED** [review]

File: pkgs/by-name/gi/git-lfs/package.nix [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        --bash <($out/bin/git-lfs completion bash) \
+        --fish <($out/bin/git-lfs completion fish) \
+        --zsh <($out/bin/git-lfs completion zsh)
+    '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgramArg = [ "--version" ];
+  doInstallCheck = true;
```
Comment (ambroisie) [inline pkgs/by-name/gi/git-lfs/package.nix:99]: Why the switch from `passthru.tests` and `testers.testVersion`?

Comment (niklaskorz) [reply pkgs/by-name/gi/git-lfs/package.nix:99]: Install checks are run as part of `nix-build`, so they are harder to miss than passthru tests.

Comment (GaetanLepage) [reply pkgs/by-name/gi/git-lfs/package.nix:99]: Yes, it is recommended over `passthru.tests.testVersion` in the docs.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**